typescript.js

```diff
@@ -9,15 +9,19 @@ module.exports = {
       parserOptions: { warnOnUnsupportedTypeScriptVersion: false },
       extends: ['plugin:@typescript-eslint/recommended'],
       rules: {
-        'no-use-before-define': 'off', // Turning off because we have the TS version off
-        '@typescript-eslint/no-use-before-define': 'off',
-        '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true }],
-
+        'no-use-before-define': 'off', // @typescript-eslint/no-use-before-defined requires this to be off
+        '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
         'no-unused-vars': 'off', // @typescript-eslint/no-unused-vars requires this to be off
-        '@typescript-eslint/no-unused-vars': ['error', { argsIgnorePattern: '^_' }],
-
-        // TS complains about using the .ts/.tsx extensions in the import statement. so turning this rule off for TS
-        'import/extensions': 'off',
+        'no-empty-functions': 'off', // @typescript-eslint/no-empty-function requires this to be off
+        '@typescript-eslint/no-empty-function': ['error', { allow: ['arrowFunctions'] }],
```

Comment:
Note from Jordi:
@typescript-eslint/no-empty-function -- as error except for arrow functions to allow for empty functions when mocking functions or making them the default for a provider. https://typescript-eslint.io/rules/no-empty-function/ We already have no-empty-function as an error, this one builds on that rule and makes it work for typescript.

Reply:
So is this just allowing a noOp function to not error out, like this one? 
```javascript
const noOp = () => {}
```

If so, I have found that it's happy if I just use this, which is more clear:
```javascript
const noOp = () => undefined
```

Reply:
Oh, that makes sense. We can remove this line and let it use the default ( "@typescript-eslint/no-empty-function": [ "error"])
For javascript files, it has this rule:
"no-empty-function": [ "error", { "allow": [ "arrowFunctions", "functions", "methods" ] } ],
Maybe that should be changed to 
"no-empty-function": ["error", {allow:[]}] // empty array for allow to override airbnb's change for this rule
 for consistency with TS?
 
Airbnb allows those 3 things here: https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/rules/best-practices.js

And you can read about what they do here: https://eslint.org/docs/latest/rules/no-empty-function
You can also just leave comments inside the empty {} to please the rule.

Reply:
I think I won't change the javascript one for now so less change happens. But I'm fine with removing the TS override and just use the default for that.
```suggestion
```